### PR TITLE
Improvements for "installing from source" and "command-line" documents.

### LINKED ISSUE #5625: Documentation for installing from Source for 2.9.0 is incomplete/perhaps incorrect
**CKAN version**
2.9.0

**Describe the bug**

Documentation issues - or possibly installation issues.

Following instructions to install 2.9.0 from source (https://docs.ckan.org/en/2.9/maintaining/installing/install-from-source.html) on a clean Ubuntu Server 20.04 VM does not lead to a working CKAN on port 5000. There are a few issues, some of which I have found work arounds for, but ultimately, no webpage is served on port 5000, and it is unclear whether to continue with deploying the server or not.

**Steps to reproduce**
Follow the steps on https://docs.ckan.org/en/2.9/maintaining/installing/install-from-source.html from a new installation of Ubuntu 20.04 Server. I am choosing the latest stable version in part 2(c), and running everything (postgres/solr) on the same single server. I am not installing anything optional (eg, datastore). See additional details below for details and suggestions on my experience so far.

**Expected behavior**
In point 9, "Open http://127.0.0.1:5000/ in a web browser, and you should see the CKAN front page." 

**Additional details**
In step (1) you say to install the solr-jetty package, but in step (5), the instructions tell you how to configure tomcat. So, do we go with Jetty or Tomcat?

First try: I stayed with solr-jetty.

In that case, I think step 5 should include jetty instructions, which I think are that you should
`sudo nano /etc/jetty9/start.ini` and replace `jetty.port=8080` with `jetty.port=8983`. I am not sure whether similar 8080->8983 changes are needed in `etc/jetty9/jetty-http.xml` or `etc/jetty9/jetty-spring.xml` where 8080 comes up, but I did so to be sure. Step 5.2 is the same - then `sudo service jetty9 restart`, before you try to browser to localhost:8983. It seems to work - you get the jetty placeholder there, and solr is correctly on localhost:8983/solr - although it looks like quite an old version.

When I finsh the steps and try `ckan -c /etc/ckan/default/ckan.ini run`,  the console reports:
```
2020-09-29 13:52:34,179 INFO  [ckan.cli] Using configuration file /etc/ckan/default/ckan.ini
2020-09-29 13:52:34,179 INFO  [ckan.config.environment] Loading static files from public
2020-09-29 13:52:34,202 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:34,339 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:34,389 CRITI [ckan.lib.uploader] Please specify a ckan.storage_path in your config
                         for your uploads
2020-09-29 13:52:34,483 INFO  [ckan.cli.server] Running server localhost on port 5000
2020-09-29 13:52:35,032 INFO  [ckan.cli] Using configuration file /etc/ckan/default/ckan.ini
2020-09-29 13:52:35,032 INFO  [ckan.config.environment] Loading static files from public
2020-09-29 13:52:35,056 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:35,195 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:35,246 CRITI [ckan.lib.uploader] Please specify a ckan.storage_path in your config
                         for your uploads
2020-09-29 13:52:35,340 INFO  [ckan.cli.server] Running server localhost on port 5000
``` 
but nothing served on port 5000 - a long delay waiting for response, and timeout.
`sudo netstat -peanut |grep 5000` (if I've installed net-tools) confirms I have
`tcp        0      0 127.0.0.1:5000          0.0.0.0:*               LISTEN      1001       61677      14256/python3`

The only thing that jumps out is the ckan.storage_path absence is marked as "CRITI" above - it would be surprising if that's the problem, since CKAN hasn't aborted, but I've set up the datastore anyway by following https://docs.ckan.org/en/2.9/maintaining/datastore.html and https://docs.ckan.org/en/2.9/maintaining/filestore.html - and the CRITI warning is now gone, but still no success on viewing a page on port 5000. (On that note, it may be good to mention the filestore vs datastore issue in Step 8 on the main installation page, rather than jumping for datastore)

Anyhow, at this point, I'm stuck and have nothing else to go on.

On my second try, I tried replacing solr-jetty with solr-tomcat in step 1, which then meant the instructions in step 5 were ok. I got exactly the same outcome though, with nothing served on 5000.

Thirdly, I read in a few places that solr-tomcat and solr-jetty both come with old versions of solr, so on the third try, I installed neither solr-jetty, nor solr-tomcat in Step 1, and replaced Step 5 with the instructions at https://github.com/ckan/ckan/wiki/Install-and-use-Solr-6.5-with-CKAN - the new Solr looks much more polished and I
could browse to it as expected on 8983 with the different solr_url, but again, nothing visible on port 5000 when I run the last step.

### PR DESCRIPTION
**Pull request by weshinsley**
Fixes #5625 and possibly also #5665 and #5662

### Proposed fixes:

These are all documentation improvements, which came up as I installed a new CKAN 2.9.0 server, so 
they have been "tested" by following them; I think they improve clarity in a couple of places.

* Installing from source - install solr-tomcat instead of solr-jetty (to be consistent with solr section)
* Installing from source - mention use of --host 0.0.0.0 in final section for testing dev server
* CLI docs - correct `ckan server` to `ckan run`
* [ADDED LATER] - mention that `ckan.site_url` must _exactly_ match what's in the browser URL - eg after ssl cert deployment.

### Features:

(This is a documentation only PR - I'm not sure whether that counts as "user-visible" changes)

- [ ] includes tests covering changes
- [X] includes updated documentation
- [X] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by weshinsley** [comment]
Hmm - the test-python-3 failure is not related to this PR.
Looks like you may have some different sorting behaviour going on when you sort by number_created_packages, but have a couple of users with the same number. I think you're expecting secondary alpha-sort but not getting it...

**Comment by Zharktas** [comment]
There is some flakyness in the failing test, sometimes it passes, sometimes it doesn't even with the same commit. It probably passes if you rerun the failing job. Most likely the used sorting is not stable so they end up in different order, but it is not related to this pr.

**Comment by weshinsley** [comment]
Thanks @Zharktas - indeed, it passed this time!

**Review by kowh-ai — APPROVED** [review]
Documentation changes all look good

File: doc/maintaining/cli.rst [hunk 1]
```diff
@@ -482,6 +482,18 @@ The result is saved in profile.data.search. To view the profile in runsnakerun::
...
+run: Start Development server
+==================================
+
+Usage
+
+.. parsed-literal::
+
+ ckan run --host (-h)          - Set Host
+ ckan run --port (-p)          - Set Port
+ ckan run --reloader (-r)      - Use reloader
```
Comment (amercader) [inline doc/maintaining/cli.rst:494]: This needs to be updated after #5639

File: doc/maintaining/installing/install-from-source.rst [hunk 2]
```diff
@@ -290,7 +289,13 @@ useful for development and testing:
     ckan -c |ckan.ini| run
 
 Open http://127.0.0.1:5000/ in a web browser, and you should see the CKAN front
-page.
+page. 
+
+.. note:: by default, the development server listens for requests only from 
+    the localhost, 127.0.0.1. If you are installing into a server os with no web browser
```
Comment (amercader) [inline doc/maintaining/installing/install-from-source.rst:295]:
```suggestion
    the localhost, 127.0.0.1. If you are installing into a server OS with no web browser 
```

**Comment by weshinsley** [comment]
@kowh-ai - thanks for the review - I've re-requested since I've added  a bit more text in the install-from-source file to additionally address issue #5665

**Review by kowh-ai — APPROVED** [review]
Apologies for the delay Wes - again these documentation changes look fine to me

**Comment by kowh-ai** [comment]
@amercader - this can be merged into master and the 2.9 branch

**Comment by pdelboca** [comment]
Closing due to inactivity and since docs has been rewritten recently. Please reopen it if you consider it is still relevant. Thanks!

**Comment by weshinsley** [comment]
Your call of course - but the changes I suggested came because of things that weren't clear when I installed CKAN, and I think they still serve a useful purpose, in that...

* You still have reference to both tomcat and jetty in the solr setup troubleshooting.
* It's still not made clear that you will have to revisit ckan.sitre_url should you later follow the "production" guide.
* The information about why http://127.0.0.1:5000/ might not work in a server setting is still useful.

**Pull request closed without merging**